Content.Client/CombatMode/ShowCombatModeIndicatorsOverlay.cs

```diff
@@ -0,0 +1,113 @@
+using Content.Shared.CCVar;
+using Content.Client.Hands.Systems;
+using Content.Shared.Weapons.Ranged.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Client.UserInterface;
+using Robust.Client.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Enums;
+using Robust.Shared.Utility;
+
+namespace Content.Client.CombatMode
+{
+    public sealed class ShowCombatModeIndicatorsOverlay : Overlay
+    {
+        [Dependency] private readonly IConfigurationManager _cfg = default!;
+        [Dependency] private readonly IInputManager _inputManager = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly IEyeManager _eye = default!;
+        private Texture? _gunSight;
+        private Texture? _meleeSight;
+        private CombatModeSystem _combatSystem;
+        private IPlayerManager _player;
+        public override OverlaySpace Space => OverlaySpace.ScreenSpace;
+        public ShowCombatModeIndicatorsOverlay(IPlayerManager playerManager, CombatModeSystem combatSys)
+        {
+            IoCManager.InjectDependencies(this);
+
+            _combatSystem = combatSys;
+            _player = playerManager;
+
+            _gunSight = GetTextureFromRsi("gunsight");
+            _meleeSight = GetTextureFromRsi("meleesight");
+        }
+
+        private Texture GetTextureFromRsi(string _spriteName)
+        {
+            var sprite = new SpriteSpecifier.Rsi(
+                new ResPath("/Textures/Interface/Misc/pointer_sights.rsi"), _spriteName);
+            return _entMan.EntitySysManager.GetEntitySystem<SpriteSystem>().Frame0(sprite);
+        }
+
+        protected override void DisposeBehavior()
+        {
+            base.DisposeBehavior();
+        }
+
+        protected override void Draw(in OverlayDrawArgs args)
+        {
+            if (!_cfg.GetCVar(CCVars.HudHeldItemShow))
+                return;
+
+            var player = _player.LocalPlayer?.ControlledEntity;
+
+            if (player == null ||
+                !_entMan.TryGetComponent<TransformComponent>(player, out var xform))
+            {
+                return;
+            }
+
+            if (!_combatSystem.IsInCombatMode(player.Value))
+                return;
```

Comment:
This should be done in beforedraw.

Reply:
i don't know but this method don't work. It just don't run.

Reply:
maybe it is broken I don't know.

Reply:
![image](https://user-images.githubusercontent.com/2144635/235513714-267f9415-4929-4a80-83e3-72d17ef23f6f.png)

https://discord.com/channels/310555209753690112/310555209753690112/1102667555279212736

Reply:
Electro has added support for it in 4a65111.

Reply:
fixed

Reply:
unfortunately this doesn't work for me. Yes, this method is run now, but it leads to errors in the next Overlay as I understand it. I do not know, but if it is not drawn(return false), then the next overlay "ShowHandItemOverlay" is not drawn too (his method beforeDraw doesn't run for this case).